dao/src/main/scala/za/co/absa/enceladus/dao/EnceladusRestDAO.scala

```diff
@@ -15,25 +15,40 @@
 
 package za.co.absa.enceladus.dao
 
-import com.typesafe.config.ConfigFactory
-import org.apache.commons.httpclient.HttpStatus
-import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet, HttpPost}
+import java.net.URI
+
+import scala.util.control.NonFatal
+
+import org.apache.http.HttpStatus
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.client.methods.HttpGet
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.impl.client.HttpClientBuilder
 import org.apache.http.impl.client.HttpClients
 import org.apache.http.util.EntityUtils
 import org.apache.log4j.LogManager
-import org.apache.spark.sql.types.{DataType, StructType}
-import za.co.absa.enceladus.model._
-
-import scala.util.control.NonFatal
-import com.fasterxml.jackson.databind.{DeserializationFeature, ObjectMapper, SerializationFeature}
-import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.apache.spark.sql.types.DataType
+import org.apache.spark.sql.types.StructType
+import org.springframework.http.{HttpStatus => SpringHttpStatus}
+import org.springframework.security.kerberos.client.KerberosRestTemplate
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.typesafe.config.ConfigFactory
+
+import za.co.absa.enceladus.model.Dataset
+import za.co.absa.enceladus.model.MappingTable
+import sun.security.krb5.internal.ktab.KeyTab
 
 object EnceladusRestDAO extends EnceladusDAO {
   val conf = ConfigFactory.load()
   val restBase = conf.getString("menas.rest.uri")
-  private val userName = LoggedInUserInfo.getUserName
-
+  
+  private var _userName: String = ""
```

Comment:
can't this be a `lazy val` instead of `var`?

Reply:
Hmmm... Maybe we should change the DAO to a class... it'd be much cleaner.

Let's raise a new one to refactor this pls.

Reply:
#492